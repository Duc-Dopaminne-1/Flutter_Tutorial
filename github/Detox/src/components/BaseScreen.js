import React from 'react';
import {StyleSheet, View} from 'react-native';
import LinearGradient from 'react-native-linear-gradient';

import {translate} from '../assets/localize';
import {STRINGS} from '../assets/localize/string';
import {COLORS} from '../assets/theme/colors';
import {HELPERS} from '../assets/theme/helpers';
import {METRICS} from '../assets/theme/metric';
import {commonStyles} from '../assets/theme/styles';
import Header from './Header';
import SafeAreaScreenContainer from './SafeAreaScreenContainer';

const styles = StyleSheet.create({
  safeArea: {
    backgroundColor: COLORS.NEUTRAL_WHITE,
  },
  container: {
    ...HELPERS.fill,
    backgroundColor: COLORS.BACKGROUND,
  },
  shadowContainer: {position: 'absolute', right: 0, left: 0, height: 20},
});

const BaseScreen = ({
  title = translate(STRINGS.BACK),
  isBackable = true,
  showHeader = true,
  isFullScreen = false,
  rightComponent,
  leftComponent,
  children,
  titleProps,
  headerOptions = null,
  onBackPress,
  containerStyle = {},
  testID = '',
  modals,
  animation,
  customStyle,
  showHeaderShadow,
}) => {
  return isFullScreen ? (
    <View style={styles.container}>{children}</View>
  ) : (
    <SafeAreaScreenContainer testID={testID} style={[styles.safeArea, customStyle]}>
      <View style={[styles.container, containerStyle]}>
        {showHeader && (
          <>
            <View
              style={[
                METRICS.horizontalPadding,
                HELPERS.zIndexView,
                {backgroundColor: COLORS.NEUTRAL_WHITE},
              ]}>
              <Header
                animation={animation}
                onBackPress={onBackPress}
                leftTextProps={titleProps}
                leftText={title}
                isBackable={isBackable}
                leftComponent={leftComponent}
                rightComponent={rightComponent}
                {...headerOptions}
              />
            </View>
          </>
        )}
        {showHeaderShadow && (
          <View style={HELPERS.zIndexView}>
            <View style={styles.shadowContainer}>
              <LinearGradient
                colors={[COLORS.BLACK_OPACITY_01, COLORS.TRANSPARENT_OPACITY]}
                style={[commonStyles.separatorRow12, {backgroundColor: COLORS.TRANSPARENT}]}
              />
            </View>
          </View>
        )}
        {children}
      </View>
      {modals}
    </SafeAreaScreenContainer>
  );
};

export default BaseScreen;
