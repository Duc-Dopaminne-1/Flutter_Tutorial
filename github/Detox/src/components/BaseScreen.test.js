import React from 'react';
import renderer from 'react-test-renderer';

import BaseScreen from './BaseScreen';

it('renders correctly', () => {
  const tree = renderer.create(<BaseScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
