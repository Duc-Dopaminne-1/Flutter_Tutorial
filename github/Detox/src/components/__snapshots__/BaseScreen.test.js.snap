// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#F9FAFB",
        "flex": 1,
      },
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
        },
        undefined,
      ],
      Object {},
    ]
  }
  testID=""
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#F9FAFB",
          "flex": 1,
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "paddingHorizontal": 16,
          },
          Object {
            "zIndex": 99,
          },
          Object {
            "backgroundColor": "#FFFFFF",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingVertical": 16,
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "zIndex": 99,
            }
          }
        >
          <View
            accessibilityLabel="BackButton"
            accessible={true}
            collapsable={false}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 20,
                "left": 20,
                "right": 20,
                "top": 20,
              }
            }
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
                "paddingEnd": 12,
                "paddingVertical": 8,
                "zIndex": 99,
              }
            }
            testID="BackButton"
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/images/arrow_left_linear.png",
                }
              }
            />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#313131",
                  "flex": 1,
                  "fontFamily": "Nunito-Bold",
                  "fontSize": 24,
                  "includeFontPadding": true,
                },
                Object {},
                null,
              ]
            }
          >
            Trở về
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        />
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
